Rakefile

```diff
@@ -448,29 +448,64 @@ end
 # set this to run in crontab
 desc "get pkg availability info"
 task :get_availability_shields  do
-#  dante
   site_config = YAML.load_file("./config.yaml")
   for reldev in ['release', 'devel']
+    puts "Working on #{reldev}"
     numeric_version = (reldev == 'release') ? site_config['release_version'] : site_config['devel_version']
-    unsupported_platforms = {}
     meat_index_file = File.join("tmp", "build_dbs","#{reldev}-bioc.meat-index.txt")
-    json_file = File.join("assets", "packages", "json", numeric_version, "bioc",
-      "packages.json")
+    json_file = File.join("assets", "packages", "json", numeric_version, "bioc", "packages.json")
+    json_obj = JSON.parse(File.read(json_file))
+    indexList=[]
     if File.exists? meat_index_file
       mitxt = File.readlines(meat_index_file).join
       meat_index = Dcf.parse(mitxt)
-      json_obj = JSON.parse(File.read(json_file))
       if (not meat_index.nil?)
-	for item in meat_index
-	  view = json_obj[item['Package']]
-	  next if view.nil?
-	  get_available(item, numeric_version, view)
```

Comment:
generally, I like this design better, where a function encapsulates an action. This makes it easier to think about isolated operations in the code, including testing the specific functionality without having to process all badges...